Moonshot/player/PlayerArsenal.gd

```diff
@@ -176,7 +176,7 @@ func get_weapon():
 
 func set_weapon(new_weapon: String = "laser_blaster") -> void:
 	current_weapon_type = new_weapon
-	Utils.Player.get_parent().add_child(get_weapon().sound)
+	Utils.Player.add_child(get_weapon().sound)
```

Comment:
Audio nodes were being added to the main gain and not being deleted. This change moves audio nodes to the player.